network/net.go

```diff
@@ -505,29 +549,47 @@ func (sc *SecureTCPConn) ServerIdentity() *ServerIdentity {
 // exchangeServerIdentity is made to exchange the ServerIdentity between the two parties.
 // when a connection request is made during listening
 func (sc *SecureTCPConn) exchangeServerIdentity() error {
-	ourEnt := sc.SecureTCPHost.entity
+	ourEnt := sc.SecureTCPHost.serverIdentity
 	if ourEnt == nil {
 		ourEnt = NewServerIdentity(config.NewKeyPair(Suite).Public, "")
 	}
 	// Send our ServerIdentity to the remote endpoint
-	log.Lvl4("Sending our identity", ourEnt.ID, "to",
+	log.Lvlf4("Sending our identity %x to %s", ourEnt.ID,
 		sc.TCPConn.conn.RemoteAddr().String())
 	if err := sc.TCPConn.Send(context.TODO(), ourEnt); err != nil {
-		return fmt.Errorf("Error while sending indentity during negotiation:%s", err)
+		return fmt.Errorf("Error while sending indentity during negotiation: %s", err)
+	}
+
+	// Try for 1 second to receive the other entity
+	var nm Packet
+	var err error
+	for i := 0; i < 10; i++ {
+		// Receive the other ServerIdentity
+		nm, err = sc.TCPConn.Receive(context.TODO())
+		switch {
+		case err == nil:
+			if i > 0 {
+				log.Warn(sc, "Got a packet after a failure")
+			}
+			i = 10
```

Comment:
break in switch doesn't break the for-loop:

https://play.golang.org/p/Po3eME1Ndy

but I'd be happy to see how to do it more clean!